modules/core/src/main/java/org/apache/ignite/internal/processors/performancestatistics/FilePerformanceStatisticsReader.java

```diff
@@ -0,0 +1,342 @@
+                walkFileTree(file.toPath(), EnumSet.noneOf(FileVisitOption.class), 1,
+                    new SimpleFileVisitor<Path>() {
+                        @Override public FileVisitResult visitFile(Path path, BasicFileAttributes attrs) {
+                            if (checkFileName(path.toFile()) != null)
+                                files.add(path.toFile());
+
+                            return FileVisitResult.CONTINUE;
+                        }
+                    });
+
+                continue;
+            }
+
+            if (checkFileName(file) != null)
+                files.add(file);
+        }
+
+        return files;
+    }
+
+    /** @return UUID node of file. {@code Null} if this is not a statistics file. */
+    @Nullable private static UUID checkFileName(File file) {
+        Matcher matcher = FILE_PATTERN.matcher(file.getName());
+
+        if (matcher.matches())
+            return UUID.fromString(matcher.group(1));
+
+        return null;
+    }
+
+    /** Reads {@link UUID} from buffer. */
+    public static UUID readUuid(ByteBuffer buf) {
+        return new UUID(buf.getLong(), buf.getLong());
+    }
+
+    /** Reads {@link IgniteUuid} from buffer. */
+    public static IgniteUuid readIgniteUuid(ByteBuffer buf) {
+        UUID globalId = new UUID(buf.getLong(), buf.getLong());
+
+        return new IgniteUuid(globalId, buf.getLong());
+    }
+
+    /** Performance statistics operations deserializer. */
+    private static class PerformanceStatisticsDeserializer {
+        /** Handlers to process deserialized operation. */
+        private final PerformanceStatisticsHandler[] handlers;
+
+        /** Node id. */
+        private final UUID nodeId;
+
+        /** @param handlers Handlers to process deserialized operation. */
+        PerformanceStatisticsDeserializer(UUID nodeId, PerformanceStatisticsHandler... handlers) {
+            this.nodeId = nodeId;
+            this.handlers = handlers;
+        }
+
+        /**
+         * Tries to deserialize performance statistics operation from buffer and notify handlers.
+         *
+         * @param buf Buffer.
+         * @return {@code True} if operation deserialized and handlers notified. {@code False} if not enough bytes.
+         */
+        boolean read(ByteBuffer buf) {
+            int pos = buf.position();
+
+            if (deserialize(buf))
+                return true;
+
+            buf.position(pos);
+
+            return false;
+        }
+
+        /**
+         * @param buf Buffer.
+         * @return {@code True} if operation deserialized. {@code False} if not enough bytes.
+         */
+        private boolean deserialize(ByteBuffer buf) {
+            if (buf.remaining() < 1)
+                return false;
+
+            byte opTypeByte = buf.get();
+
+            OperationType opType = OperationType.fromOrdinal(opTypeByte);
+
+            if (cacheOperation(opType)) {
+                if (buf.remaining() < 4 + 8 + 8)
```

Comment:
We have code to calculate record size both in the writer and in the reader.
Let's have it in one place - `OperationType`.